lib/Sema/TypeCheckProtocol.cpp

```diff
@@ -1430,29 +1431,17 @@ bool WitnessChecker::findBestWitness(
 }
 
 bool WitnessChecker::
-checkWitnessAccessibility(const DeclContext *&requiredAccessScope,
+checkWitnessAccessibility(AccessScope &requiredAccessScope,
                           ValueDecl *requirement,
                           ValueDecl *witness,
                           bool *isSetter) {
   *isSetter = false;
 
-  const DeclContext *protoAccessScope = Proto->getFormalAccessScope(DC);
+  requiredAccessScope =
+    *requiredAccessScope.intersectWith(Proto->getFormalAccessScope(DC));
```

Comment:
It is needed here, because`intersectWith` returns `Optional`.

Previously here was an assert for "non-intersecting scopes" case, now it will crash in the same scenario. Should I make an explicit assert?